src/api/users/src/models/schema.js

```diff
@@ -0,0 +1,54 @@
+const { createError, hash } = require('@senecacdot/satellite');
+const { celebrate, Joi, Segments } = require('celebrate');
+
+const validatePagingParams = () =>
+  celebrate({
+    [Segments.QUERY]: {
+      per_page: Joi.number().integer().min(1).max(100).default(20),
+      start_after: Joi.string().hex(),
+    },
+  });
+
+const validateId = () =>
+  celebrate({
+    [Segments.PARAMS]: {
+      id: Joi.string().hex().required(),
+    },
+  });
+
+// Generate a display name if none given
+const generateDisplayName = (parent) => `${parent.firstName} ${parent.lastName}`;
+
+const validateUser = () =>
+  celebrate({
+    [Segments.BODY]: Joi.object().keys({
+      firstName: Joi.string().required(),
+      lastName: Joi.string().required(),
+      email: Joi.string().email().required(),
+      displayName: Joi.string().default(generateDisplayName),
+      isAdmin: Joi.boolean().default(false),
+      isFlagged: Joi.boolean().default(false),
+      feeds: Joi.array().items(Joi.string().uri()).required(),
+      github: Joi.object({
+        username: Joi.string(),
+        avatarUrl: Joi.string().uri(),
+      }).and('username', 'avatarUrl'),
+    }),
+  });
+
+// Make sure that the :id param matches the hashed value of the user's email if present
+const validateEmailHash = () => (req, res, next) => {
+  const { email } = req.body;
+  const { id } = req.params;
+
+  if (id !== hash(email)) {
```

Comment:
This should have a comment specifying that the hash function in satellite only returns the first 10 characters of the generated hash. The id passed in the post request must be 10 characters long.

Reply:
Fixed in https://github.com/Seneca-CDOT/telescope/pull/2060/commits/681c129eef9a5647893c52d152b43903186a5936.